ddtrace/tracer/servicehash.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package tracer
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"sync"
+)
+
+const maxCacheSize = 1000
+
+var (
+	cache = make(map[string]string)
+	lock  sync.RWMutex
+)
+
+// Hash computes hash for `service`. Will use a local cache to avoid recomputation.
+func Hash(service string) string {
+	hash := getHashFromCache(service)
+	if len(hash) > 0 {
+		return hash
+	}
+	hashb := sha256.Sum256([]byte(service))
+	// Only grab first 10 characters (2 chars per byte * 5 bytes = 10 chars)
```

Comment:
This is confusing, you're slicing at 5 below

Reply:
Hmm any ideas on how to make this clearer? I added the comment to clarify that we want 10 characters of hash and since each byte becomes 2 hex characters we need to slice only the first 5 bytes of hash data to end up with a 10 character long hex string

Reply:
It's clear now. Thanks 🙂